src/client/io_tracker.cpp

```diff
@@ -84,10 +85,15 @@ void IOTracker::StartAioRead(CurveAioContext* ctx, MDSClient* mdsclient,
     DVLOG(9) << "aioread op, offset = " << ctx->offset
              << ", length = " << ctx->length;
 
-    DoRead(mdsclient, fileInfo);
+    DoRead(mdsclient, fileInfo, throttle);
 }
 
-void IOTracker::DoRead(MDSClient* mdsclient, const FInfo_t* fileInfo) {
+void IOTracker::DoRead(MDSClient* mdsclient, const FInfo_t* fileInfo,
+                       Throttle* throttle) {
+    if (throttle) {
+        throttle->Get(true, length_);
```

Comment:
Will blocking here cause a deadlock ？

Reply:
If tokens is not enough, this call will block until tokens is available. And, when close a file, in throttle's destructor it will let a blocked requests pass.